src/plugins/vis_type_vislib/public/to_ast.ts

```diff
@@ -88,12 +88,11 @@ export const toExpressionAst: VisToExpressionAst<BasicVislibParams> = async (vis
 
   visConfig.dimensions = dimensions;
 
-  const configStr = JSON.stringify(visConfig).replace(/\\/g, `\\\\`).replace(/'/g, `\\'`);
   const visTypeXy = buildExpressionFunction<VisTypeVislibExpressionFunctionDefinition>(
     vislibVisName,
     {
       type: vis.type.name,
-      visConfig: configStr,
```

Comment:
There was a regression here which AFAICT was introduced by https://github.com/elastic/kibana/pull/80744.

It surfaced as a bug where creating a vislib vis with a serial diff agg threw an error... Because the custom label that got created had incorrectly double-escaped backslashes, e.g.: `"Blah blah serial diff of \\"bytes\\""`

This would cause `JSON.parse` to throw because the double-escaped quotes were accidentally kept in place and interpreted as part of the json, unexpectly ending the line early.

I know this escaping was copied over from `build_pipeline` and TBH I'm not 100% sure why it isn't needed anymore, but thought I'd provide some background. I only found this bug because I was testing visualizations using every agg type to validate this PR, and ran into trouble with serial diff.

Also worth pointing out that the `to_ast` for pie charts still has the escaping in place, because i couldn't find a situation where it was broken. Might be worth testing that further to verify.

Reply:
I tested both and I think that they both work as they should. I tested all the aggs on all vislib visualizations. Can't find any regressions.